base/loading.jl

```diff
@@ -566,7 +566,12 @@ end
 ## generic project & manifest API ##
 
 const project_names = ("JuliaProject.toml", "Project.toml")
-const manifest_names = ("JuliaManifest.toml", "Manifest.toml")
+const manifest_names = (
+    "JuliaManifest.$(VERSION.major).$(VERSION.minor).toml",
+    "Manifest.$(VERSION.major).$(VERSION.minor).toml",
+    "JuliaManifest.toml",
+    "Manifest.toml",
+)
```

Comment:
I guess we'll want `Manifest.$(VERSION.major).toml` etc. too, to soften any transition in a major julia release.